### ISSUE TITLE
Upgrade to Opal 1.6+

### ISSUE THREAD

**Issue by michaeljb** [tobymao/18xx#9608]
* the first attempt to upgrade to Opal 1.6 caused some view errors and was quickly rolled back
* more investigation is needed to confirm, but the second attempt (yesterday) is possibly the cause of #9603 and #9605, and maybe some slowness discussed in chat:

> 23:42:49 Is it just my impression or have games been taking much longer to load in the last ~18hrs?
> 23:44:16 games have stalled loading for me a few times today, yeah
> 00:24:44 Yep, definitely slower for me. 

I've rolled back to Opal 1.5. I'll be able to investigate in 1.5-2 weeks.

update 9/24 - creating TODO list out of the above items

TODO:
- [x]  1822 passing issues
    - [#9603](https://github.com/tobymao/18xx/issues/9603) (canonical) - 1822 - https://18xx.games/game/134278?action=210
    - [#9604](https://github.com/tobymao/18xx/issues/9604) - 1822PNW - https://18xx.games/game/134371?action=224
    - [#9606](https://github.com/tobymao/18xx/issues/9606) - 1822Africa - https://18xx.games/game/134262?action=96, or possibly Kirasjer87's earlier turn in the SR
- [x] [#9605](https://github.com/tobymao/18xx/issues/9605) - map disappearing when clicking on a tile with no available upgrades
- [x] investigate slowness
- [ ] investigate mega slowness / hanging in Chrome

**Comment by michaeljb** [comment]
Confirmed that rolling back to 1.5 fixed #9605

**Comment by michaeljb** [comment]
Rolling back seems to have fixed #9603

**Comment by michaeljb** [comment]
> - [ ] investigate slowness

I've set up two local instances with a timer around `@game = Engine::Game.load( ... )` in `game_page.rb`, one based on `master` and one with Opal 1.6 instead of 1.5.

Loading [this game](https://gist.github.com/michaeljb/16d6735fa8dc6c3cd97397e253c2602e) in hotseat mode is consistently slower with Opal 1.6. I threw a few of the values from Firefox into a spreadsheet, 1.6 is averaging **2.8 seconds** to load the game, while 1.5 is averaging **1.9 seconds**. I'm seeing a similar disparity in Chrome, but on occasion with 1.6 the game doesn't finish loading even after 30 seconds.

More investigation needed, but at least I've repro'd a clear performance drop with Opal 1.6...

**Comment by michaeljb** [comment]
I verified via bisecting that [opal/opal #2357](https://www.github.com/opal/opal/pull/2357) is the cause of the performance hit, and the description and some of the comments on that PR support this conclusion. If we go with Opal 1.6, I don't think this is something we can avoid. (now that I'm typing this up, I realize I haven't looked into the 30+ seconds of hanging in Chrome)

I did find another performance issue--it only affects game classes which depend on other game classes (e.g., 1822MRS depending on 1822), but it would have made performance hits from Opal 1.6 more noticeable.
